src/app/medicine/medicine.cpp

```diff
@@ -0,0 +1,26 @@
+#include "medicine.hpp"
+
+Medicine::Medicine(const std::string &name, const double price, const int amount, const int expirationDate,
+                   const std::pair<double, double> &storageTemperature, const std::vector<std::string> &composition)
+    : Product(name, price, amount), expirationDate_(expirationDate), storageTemperature_(storageTemperature),
+      composition_(composition)
+{
+}
+bool Medicine::checkIfExpired()
+{
+    // TODO: Write this function when Date will be ready
+    return false;
+}
+
+std::vector<std::pair<std::string, std::string>> Medicine::getInfo() const
```

Comment:
I think two best options exist for us, than returning vector<string>:
1. Stick to seperate getters (getExpirationDate(), getStorageTemperature etc.)
2. Return structure containing: expiration, storage temperature, vector<string> with compositions.

I don't know which is better for us, but both are easier to maintain and less error prone.

Here we stick to fixed string structure very dificult to parse and keep aligned with setter:
1. If we change anything in return we might broke a lot of code using getInfo()
2. Every time somebody calls getInfo, he/she/them/they (or whatever) will have to study structure of returned to string to extract a piece of interesting information. And can easily get confused.

Reply:
Because we use pointers to Product, those getters must be placed as virtual functions in Product class and also in Medicine class and it leads to creation of unnecessary code in those classes.

Reply:
Adjustment to std::map

Reply:
Done